### PR TITLE
Multiprocess improvements

### PR DESCRIPTION
**Pull request by nithinraok**
# What does this PR do ?

Improve multiprocessing speed by reading audio file only once, and also remove type_check decorator for forward_to_export function in label_models.py 

**Collection**: ASR

**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

File: scripts/speaker_tasks/scp_to_manifest.py [hunk 1]
```diff
@@ -146,52 +173,28 @@ def get_labels(lines):
...
+        out_file = os.path.splitext(scp)[0] + '_manifest.json'
+        write_file(out_file, lines, range(len(lines)))
     else:
         lines = read_manifest(manifest)
 
     lines = process_map(get_duration, lines, chunksize=100)
 
     if create_chunks:
-        print("creating chunk")
+        print("creating and writing chunks to {}".format(CWD))
```
Comment (tango4j) [inline scripts/speaker_tasks/scp_to_manifest.py:190]: https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md
10. F-strings are preferred to formatted strings.

I recently figured that you have already used lots of str.format() method here and there.  Following the NeMo contributing guide, I suggest you to use f-string. Which makes -> `print(f"creating and writing chunks to {CWD}"`

str.format() appeared in python 2.6.

File: scripts/speaker_tasks/scp_to_manifest.py [hunk 2]
```diff
@@ -31,46 +30,74 @@
 This scipt converts a scp file where each line contains  
 <absolute path of wav file> 
 to a manifest json file. 
+Optionally post processes the manifest file to create dev and train split for speaker embedding 
+training, also optionally chunk an audio file in to segments of random DURATIONS and create those
+wav files in CWD. 
+
+While creating chunks, if audio is not sampled at 16Khz, it resamples to 16Khz and write the wav file.
 Args: 
 --scp: scp file name
```
Comment (tango4j) [inline scripts/speaker_tasks/scp_to_manifest.py:39]: I think this is a good time for doing "scp-> filelist" refactor, although you need to change the name of the file.
Two problems: 
(1) What you are using is not a SCP file.
(2) Even if it is SCP file, SCP file format is very hard to find the description and there is no explanation about scp file format in NeMo. 

I believe this script is quite important to prepare dataset for training speaker embedding model. I think changing the name is necessary.

**Review by tango4j — CHANGES_REQUESTED** [review]

File: scripts/speaker_tasks/scp_to_manifest.py (hunk 1 above)
Comment (nithinraok) [reply scripts/speaker_tasks/scp_to_manifest.py]: Thanks tj, updated the code to use f-string.

File: scripts/speaker_tasks/scp_to_manifest.py (hunk 2 above)
Comment (nithinraok) [reply scripts/speaker_tasks/scp_to_manifest.py]: Updated

File: nemo/collections/asr/models/label_models.py [hunk 3]
```diff
@@ -251,7 +251,6 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
             "embs": NeuralType(('B', 'D'), AcousticEncodedRepresentation()),
         }
 
-    @typecheck()
```
Comment (titu1994) [inline nemo/collections/asr/models/label_models.py:254]: You can just replace your forwarded last two lines with calling forward_for_export() there. Do that in a later pr

**Review by titu1994 — COMMENTED** [review]
I've only skimmed the PR but it seems fine. Finish resolving all of @tango4j requirements then merge v

**Review by tango4j — APPROVED** [review]

**Pull request merged**